pkg/apis/duck/v1beta1/delivery_types.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+// DeliverySpec contains the delivery options for event senders,
+// such as channelable and source.
+type DeliverySpec struct {
+	// DeadLetterSink is the sink receiving event that couldn't be sent to
+	// a destination.
+	// +optional
+	DeadLetterSink *duckv1.Destination `json:"deadLetterSink,omitempty"`
```

Comment:
should be `deadLetterSinkUri`

Reply:
It's the Destination, so I don't think it actually should be an uri here?